hw/lib/signup/sign_up_screen.dart

```diff
@@ -0,0 +1,313 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:hw/gen/assets.gen.dart';
+import 'package:hw/generated/l10n.dart';
+import 'package:hw/signup/bloc/sign_up_cubit.dart';
+import 'package:hw/postSignUp/linkedIn_screen.dart';
+
+class MyHome extends StatelessWidget {
+  const MyHome({super.key});
+
+  @override
+  Widget build(BuildContext context) =>
+      MediaQuery(
+        data: MediaQuery.of(context).copyWith(
+            textScaler: TextScaler.noScaling
+        ),
+        child: BlocProvider<SignUpCubit>(
+          create: (context) => SignUpCubit(),
+          child: _HomeContent(),
+        ),
+      );
+}
+class _HomeContent extends StatelessWidget {
+  _HomeContent({super.key});
+
+  String currName = '';
+  String currEmail = '';
+  String currPassword = '';
+
+  @override
+  Widget build(BuildContext context) =>
+      BlocListener<SignUpCubit, SignUpState>(
+        listener: (context, state) {
+          if (state is SignUpFailed && state.provider =='LinkedIn')  {
```

Comment:
oh noooooooo
Logic is correct, but it's still wrong. 

we should use different state names to indicate different state of the program. In this case, the state name `SignUpFailed` indicates that sign up has failed and so the UI should follow up with error handling. 

To indicate that a sign up has success and we can move to the next step, we need to create a new state, suggest name to be `SignUpSuccess`, then we can listen to that state here and proceed to next step